pkg/config/patch_test.go

```diff
@@ -0,0 +1,148 @@
+			kind:   "Deployment",
+			object: applyappsv1.Deployment("test", "test"),
+			patches: []v1alpha1.Patch{{
+				Kind: "Deployment",
+				Patch: json.RawMessage(`
+metadata:
+  labels:
+    added: via-patch`),
+			}},
+			want: applyappsv1.Deployment("test", "test").WithLabels(map[string]string{"added": "via-patch"}),
+		},
+		{
+			name:   "add labels to deployment (smp, json)",
+			kind:   "Deployment",
+			object: applyappsv1.Deployment("test", "test"),
+			patches: []v1alpha1.Patch{{
+				Kind: "Deployment",
+				Patch: json.RawMessage(`{
+                  "metadata": {
+                    "labels": {
+                      "added": "via-patch"
+                    }
+                  }
+                }`),
+			}},
+			want: applyappsv1.Deployment("test", "test").WithLabels(map[string]string{"added": "via-patch"}),
+		},
+		{
+			name:   "add labels to deployment (JSON6902, yaml)",
+			kind:   "Deployment",
+			object: applyappsv1.Deployment("test", "test"),
+			patches: []v1alpha1.Patch{{
+				Kind: "Deployment",
+				Patch: json.RawMessage(`
+  op: add
+  path: /metadata/labels
+  value: 
+    added: via-patch`),
+			}},
+			want: applyappsv1.Deployment("test", "test").WithLabels(map[string]string{"added": "via-patch"}),
+		},
+		{
+			name:   "add labels to deployment (JSON6902, json)",
+			kind:   "Deployment",
+			object: applyappsv1.Deployment("test", "test"),
+			patches: []v1alpha1.Patch{{
+				Kind:  "Deployment",
+				Patch: json.RawMessage(`{"op": "add", "path": "/metadata/labels", "value":{"added":"via-patch"}}`),
+			}},
+			want: applyappsv1.Deployment("test", "test").WithLabels(map[string]string{"added": "via-patch"}),
+		},
+		{
+			name:   "add labels to deployment without affecting existing labels (JSON6902, json)",
+			kind:   "Deployment",
+			object: applyappsv1.Deployment("test", "test").WithLabels(map[string]string{"initial": "label"}),
+			patches: []v1alpha1.Patch{{
+				Kind:  "Deployment",
+				Patch: json.RawMessage(`{"op": "add", "path": "/metadata/labels", "value":{"added":"via-patch"}}`),
+			}},
+			want: applyappsv1.Deployment("test", "test").WithLabels(
+				map[string]string{
+					"initial": "label",
+					"added":   "via-patch",
+				},
+			),
+		},
+		{
+			name:   "add labels to deployment without affecting existing labels (smp, json)",
+			kind:   "Deployment",
+			object: applyappsv1.Deployment("test", "test").WithLabels(map[string]string{"initial": "label"}),
+			patches: []v1alpha1.Patch{{
+				Kind: "Deployment",
+				Patch: json.RawMessage(`{
+                  "metadata": {
+                    "labels": {
+                      "added": "via-patch"
+                    }
+                  }
+                }`),
+			}},
+			want: applyappsv1.Deployment("test", "test").WithLabels(
+				map[string]string{
+					"initial": "label",
+					"added":   "via-patch",
+				},
+			),
+		},
+		{
+			name:   "add labels to deployment with a wildcard match (smp, yaml)",
```

Comment:
can we add a patch that does not specify its `kind`?

Reply:
Added a lot more tests here